bls opcode name doesn't match at '<name>'

repro:

```
AcceleratorRange.EightG;
AcceleratorRange.FourG;
AcceleratorRange.OneG;
AcceleratorRange.TwoG;
AnalogPin.P0;
AnalogPin.P10;
AnalogPin.P1;
AnalogPin.P2;
AnalogPin.P3;
AnalogPin.P4;
BaudRate.BaudRate115200;
BaudRate.BaudRate9600;
BeatFraction.Eighth;
BeatFraction.Half;
BeatFraction.Quarter;
BeatFraction.Sixteenth;
BeatFraction.Whole;
Button.A;
Button.AB;
Button.B;
DAL.ANIMATION_MODE_ANIMATE_IMAGE;
DAL.ANIMATION_MODE_ANIMATE_IMAGE_WITH_CLEAR;
DAL.ANIMATION_MODE_NONE;
DAL.ANIMATION_MODE_PRINT_CHARACTER;
DAL.ANIMATION_MODE_PRINT_TEXT;
DAL.ANIMATION_MODE_SCROLL_IMAGE;
DAL.ANIMATION_MODE_SCROLL_TEXT;
DAL.ANIMATION_MODE_STOPPED;
DAL.ASYNC;
DAL.CREATE_AND_FIRE;
DAL.CREATE_ONLY;
DAL.DISPLAY_MODE_BLACK_AND_WHITE;
DAL.DISPLAY_MODE_BLACK_AND_WHITE_LIGHT_SENSE;
DAL.DISPLAY_MODE_GREYSCALE;
DAL.IO_STATUS_ANALOG_IN;
DAL.IO_STATUS_ANALOG_OUT;
DAL.IO_STATUS_DIGITAL_IN;
DAL.IO_STATUS_DIGITAL_OUT;
DAL.IO_STATUS_EVENT_ON_EDGE;
DAL.IO_STATUS_EVENT_PULSE_ON_EDGE;
DAL.IO_STATUS_TOUCH_IN;
DAL.MAG3110_DEFAULT_ADDR;
DAL.MAG3110_SAMPLE_RATES;
DAL.MAG3110_WHOAMI_VAL;
DAL.MAG_CTRL_REG1;
DAL.MAG_CTRL_REG2;
DAL.MAG_DIE_TEMP;
DAL.MAG_DR_STATUS;
DAL.MAG_OFF_X_LSB;
DAL.MAG_OFF_X_MSB;
DAL.MAG_OFF_Y_LSB;
DAL.MAG_OFF_Y_MSB;
DAL.MAG_OFF_Z_LSB;
DAL.MAG_OFF_Z_MSB;
DAL.MAG_OUT_X_LSB;
DAL.MAG_OUT_X_MSB;
DAL.MAG_OUT_Y_LSB;
DAL.MAG_OUT_Y_MSB;
DAL.MAG_OUT_Z_LSB;
DAL.MAG_OUT_Z_MSB;
DAL.MAG_SYSMOD;
DAL.MAG_WHOAMI;
DAL.MESSAGE_BUS_LISTENER_BUSY;
DAL.MESSAGE_BUS_LISTENER_DELETING;
DAL.MESSAGE_BUS_LISTENER_DROP_IF_BUSY;
DAL.MESSAGE_BUS_LISTENER_METHOD;
DAL.MESSAGE_BUS_LISTENER_NONBLOCKING;
DAL.MESSAGE_BUS_LISTENER_PARAMETERISED;
DAL.MESSAGE_BUS_LISTENER_QUEUE_IF_BUSY;
DAL.MESSAGE_BUS_LISTENER_REENTRANT;
DAL.MESSAGE_BUS_LISTENER_URGENT;
DAL.MES_ALERTS_ID;
DAL.MES_ALERT_EVT_ALARM1;
DAL.MES_ALERT_EVT_ALARM2;
DAL.MES_ALERT_EVT_ALARM3;
DAL.MES_ALERT_EVT_ALARM4;
DAL.MES_ALERT_EVT_ALARM5;
DAL.MES_ALERT_EVT_ALARM6;
DAL.MES_ALERT_EVT_DISPLAY_TOAST;
DAL.MES_ALERT_EVT_FIND_MY_PHONE;
DAL.MES_ALERT_EVT_PLAY_RINGTONE;
DAL.MES_ALERT_EVT_PLAY_SOUND;
DAL.MES_ALERT_EVT_VIBRATE;
DAL.MES_BROADCAST_GENERAL_ID;
DAL.MES_CAMERA_EVT_LAUNCH_PHOTO_MODE;
DAL.MES_CAMERA_EVT_LAUNCH_VIDEO_MODE;
DAL.MES_CAMERA_EVT_START_VIDEO_CAPTURE;
DAL.MES_CAMERA_EVT_STOP_PHOTO_MODE;
DAL.MES_CAMERA_EVT_STOP_VIDEO_CAPTURE;
DAL.MES_CAMERA_EVT_STOP_VIDEO_MODE;
DAL.MES_CAMERA_EVT_TAKE_PHOTO;
DAL.MES_CAMERA_EVT_TOGGLE_FRONT_REAR;
DAL.MES_CAMERA_ID;
DAL.MES_DEVICE_DISPLAY_OFF;
DAL.MES_DEVICE_DISPLAY_ON;
DAL.MES_DEVICE_GESTURE_DEVICE_SHAKEN;
DAL.MES_DEVICE_GESTURE_NONE;
DAL.MES_DEVICE_INCOMING_CALL;
DAL.MES_DEVICE_INCOMING_MESSAGE;
DAL.MES_DEVICE_INFO_ID;
DAL.MES_DEVICE_ORIENTATION_LANDSCAPE;
DAL.MES_DEVICE_ORIENTATION_PORTRAIT;
DAL.MES_DPAD_BUTTON_1_DOWN;
DAL.MES_DPAD_BUTTON_1_UP;
DAL.MES_DPAD_BUTTON_2_DOWN;
DAL.MES_DPAD_BUTTON_2_UP;
DAL.MES_DPAD_BUTTON_3_DOWN;
DAL.MES_DPAD_BUTTON_3_UP;
DAL.MES_DPAD_BUTTON_4_DOWN;
DAL.MES_DPAD_BUTTON_4_UP;
DAL.MES_DPAD_BUTTON_A_DOWN;
DAL.MES_DPAD_BUTTON_A_UP;
DAL.MES_DPAD_BUTTON_B_DOWN;
DAL.MES_DPAD_BUTTON_B_UP;
DAL.MES_DPAD_BUTTON_C_DOWN;
DAL.MES_DPAD_BUTTON_C_UP;
DAL.MES_DPAD_BUTTON_D_DOWN;
DAL.MES_DPAD_BUTTON_D_UP;
DAL.MES_DPAD_CONTROLLER_ID;
DAL.MES_REMOTE_CONTROL_EVT_FORWARD;
DAL.MES_REMOTE_CONTROL_EVT_NEXTTRACK;
DAL.MES_REMOTE_CONTROL_EVT_PAUSE;
DAL.MES_REMOTE_CONTROL_EVT_PLAY;
DAL.MES_REMOTE_CONTROL_EVT_PREVTRACK;
DAL.MES_REMOTE_CONTROL_EVT_REWIND;
DAL.MES_REMOTE_CONTROL_EVT_STOP;
DAL.MES_REMOTE_CONTROL_EVT_VOLUMEDOWN;
DAL.MES_REMOTE_CONTROL_EVT_VOLUMEUP;
DAL.MES_REMOTE_CONTROL_ID;
DAL.MES_SIGNAL_STRENGTH_EVT_FOUR_BAR;
DAL.MES_SIGNAL_STRENGTH_EVT_NO_BAR;
DAL.MES_SIGNAL_STRENGTH_EVT_ONE_BAR;
DAL.MES_SIGNAL_STRENGTH_EVT_THREE_BAR;
DAL.MES_SIGNAL_STRENGTH_EVT_TWO_BAR;
DAL.MES_SIGNAL_STRENGTH_ID;
DAL.MICROBIT_ACCELEROMETER_3G_TOLERANCE;
DAL.MICROBIT_ACCELEROMETER_6G_TOLERANCE;
DAL.MICROBIT_ACCELEROMETER_8G_TOLERANCE;
DAL.MICROBIT_ACCELEROMETER_EVT_3G;
DAL.MICROBIT_ACCELEROMETER_EVT_6G;
DAL.MICROBIT_ACCELEROMETER_EVT_8G;
DAL.MICROBIT_ACCELEROMETER_EVT_DATA_UPDATE;
DAL.MICROBIT_ACCELEROMETER_EVT_FACE_DOWN;
DAL.MICROBIT_ACCELEROMETER_EVT_FACE_UP;
DAL.MICROBIT_ACCELEROMETER_EVT_FREEFALL;
DAL.MICROBIT_ACCELEROMETER_EVT_NONE;
DAL.MICROBIT_ACCELEROMETER_EVT_SHAKE;
DAL.MICROBIT_ACCELEROMETER_EVT_TILT_DOWN;
DAL.MICROBIT_ACCELEROMETER_EVT_TILT_LEFT;
DAL.MICROBIT_ACCELEROMETER_EVT_TILT_RIGHT;
DAL.MICROBIT_ACCELEROMETER_EVT_TILT_UP;
DAL.MICROBIT_ACCELEROMETER_FREEFALL_TOLERANCE;
DAL.MICROBIT_ACCELEROMETER_GESTURE_DAMPING;
DAL.MICROBIT_ACCELEROMETER_REST_TOLERANCE;
DAL.MICROBIT_ACCELEROMETER_SHAKE_COUNT_THRESHOLD;
DAL.MICROBIT_ACCELEROMETER_SHAKE_DAMPING;
DAL.MICROBIT_ACCELEROMETER_SHAKE_RTX;
DAL.MICROBIT_ACCELEROMETER_SHAKE_TOLERANCE;
DAL.MICROBIT_ACCELEROMETER_TILT_TOLERANCE;
DAL.MICROBIT_ACCEL_ADDED_TO_IDLE;
DAL.MICROBIT_ACCEL_PITCH_ROLL_VALID;
DAL.MICROBIT_BLE_EVT_CONNECTED;
DAL.MICROBIT_BLE_EVT_DISCONNECTED;
DAL.MICROBIT_BLE_MAXIMUM_BONDS;
DAL.MICROBIT_BLE_MAXIMUM_SCROLLTEXT;
DAL.MICROBIT_BLE_PAIRING_TIMEOUT;
DAL.MICROBIT_BLE_PAIR_COMPLETE;
DAL.MICROBIT_BLE_PAIR_PASSCODE;
DAL.MICROBIT_BLE_PAIR_REQUEST;
DAL.MICROBIT_BLE_PAIR_SUCCESSFUL;
DAL.MICROBIT_BLE_POWER_LEVELS;
DAL.MICROBIT_BUSY;
DAL.MICROBIT_BUTTON_ALL_EVENTS;
DAL.MICROBIT_BUTTON_DOUBLE_CLICK_THRESH;
DAL.MICROBIT_BUTTON_EVT_CLICK;
DAL.MICROBIT_BUTTON_EVT_DOUBLE_CLICK;
DAL.MICROBIT_BUTTON_EVT_DOWN;
DAL.MICROBIT_BUTTON_EVT_HOLD;
DAL.MICROBIT_BUTTON_EVT_LONG_CLICK;
DAL.MICROBIT_BUTTON_EVT_UP;
DAL.MICROBIT_BUTTON_HOLD_TIME;
DAL.MICROBIT_BUTTON_LONG_CLICK_TIME;
DAL.MICROBIT_BUTTON_SIGMA_MAX;
DAL.MICROBIT_BUTTON_SIGMA_MIN;
DAL.MICROBIT_BUTTON_SIGMA_THRESH_HI;
DAL.MICROBIT_BUTTON_SIGMA_THRESH_LO;
DAL.MICROBIT_BUTTON_SIMPLE_EVENTS;
DAL.MICROBIT_BUTTON_STATE;
DAL.MICROBIT_BUTTON_STATE_CLICK;
DAL.MICROBIT_BUTTON_STATE_HOLD_TRIGGERED;
DAL.MICROBIT_BUTTON_STATE_LONG_CLICK;
DAL.MICROBIT_CALIBRATION_IN_PROGRESS;
DAL.MICROBIT_CALIBRATION_REQUIRED;
DAL.MICROBIT_CANCELLED;
DAL.MICROBIT_COMPASS_EVT_CALIBRATE;
DAL.MICROBIT_COMPASS_EVT_CAL_END;
DAL.MICROBIT_COMPASS_EVT_CAL_REQUIRED;
DAL.MICROBIT_COMPASS_EVT_CAL_START;
DAL.MICROBIT_COMPASS_EVT_CONFIG_NEEDED;
DAL.MICROBIT_COMPASS_EVT_DATA_UPDATE;
DAL.MICROBIT_COMPASS_STATUS_ADDED_TO_IDLE;
DAL.MICROBIT_COMPASS_STATUS_CALIBRATED;
DAL.MICROBIT_COMPASS_STATUS_CALIBRATING;
DAL.MICROBIT_COMPONENT_RUNNING;
DAL.MICROBIT_DEFAULT_PWM_PERIOD;
DAL.MICROBIT_DFU_HISTOGRAM_HEIGHT;
DAL.MICROBIT_DFU_HISTOGRAM_WIDTH;
DAL.MICROBIT_DFU_OPCODE_START_DFU;
DAL.MICROBIT_DISPLAY_ANIMATE_DEFAULT_POS;
DAL.MICROBIT_DISPLAY_DEFAULT_AUTOCLEAR;
DAL.MICROBIT_DISPLAY_EVT_ANIMATION_COMPLETE;
DAL.MICROBIT_DISPLAY_EVT_FREE;
DAL.MICROBIT_DISPLAY_EVT_LIGHT_SENSE;
DAL.MICROBIT_DISPLAY_GREYSCALE_BIT_DEPTH;
DAL.MICROBIT_DISPLAY_HEIGHT;
DAL.MICROBIT_DISPLAY_ROTATION_0;
DAL.MICROBIT_DISPLAY_ROTATION_180;
DAL.MICROBIT_DISPLAY_ROTATION_270;
DAL.MICROBIT_DISPLAY_ROTATION_90;
DAL.MICROBIT_DISPLAY_SPACING;
DAL.MICROBIT_DISPLAY_WIDTH;
DAL.MICROBIT_EVT_ANY;
DAL.MICROBIT_FIBER_FLAG_CHILD;
DAL.MICROBIT_FIBER_FLAG_DO_NOT_PAGE;
DAL.MICROBIT_FIBER_FLAG_FOB;
DAL.MICROBIT_FIBER_FLAG_PARENT;
DAL.MICROBIT_FONT_ASCII_END;
DAL.MICROBIT_FONT_ASCII_START;
DAL.MICROBIT_FONT_HEIGHT;
DAL.MICROBIT_FONT_WIDTH;
DAL.MICROBIT_HEAP_BLOCK_FREE;
DAL.MICROBIT_HEAP_ERROR;
DAL.MICROBIT_I2C_ERROR;
DAL.MICROBIT_I2C_MAX_RETRIES;
DAL.MICROBIT_ID_ACCELEROMETER;
DAL.MICROBIT_ID_ANY;
DAL.MICROBIT_ID_BLE;
DAL.MICROBIT_ID_BLE_UART;
DAL.MICROBIT_ID_BUTTON_A;
DAL.MICROBIT_ID_BUTTON_AB;
DAL.MICROBIT_ID_BUTTON_B;
DAL.MICROBIT_ID_BUTTON_RESET;
DAL.MICROBIT_ID_COMPASS;
DAL.MICROBIT_ID_DISPLAY;
DAL.MICROBIT_ID_GESTURE;
DAL.MICROBIT_ID_IO_P0;
DAL.MICROBIT_ID_IO_P10;
DAL.MICROBIT_ID_IO_P11;
DAL.MICROBIT_ID_IO_P12;
DAL.MICROBIT_ID_IO_P13;
DAL.MICROBIT_ID_IO_P14;
DAL.MICROBIT_ID_IO_P15;
DAL.MICROBIT_ID_IO_P16;
DAL.MICROBIT_ID_IO_P19;
DAL.MICROBIT_ID_IO_P1;
DAL.MICROBIT_ID_IO_P20;
DAL.MICROBIT_ID_IO_P2;
DAL.MICROBIT_ID_IO_P3;
DAL.MICROBIT_ID_IO_P4;
DAL.MICROBIT_ID_IO_P5;
DAL.MICROBIT_ID_IO_P6;
DAL.MICROBIT_ID_IO_P7;
DAL.MICROBIT_ID_IO_P8;
DAL.MICROBIT_ID_IO_P9;
DAL.MICROBIT_ID_MESSAGE_BUS_LISTENER;
DAL.MICROBIT_ID_MULTIBUTTON_ATTACH;
DAL.MICROBIT_ID_NOTIFY;
DAL.MICROBIT_ID_NOTIFY_ONE;
DAL.MICROBIT_ID_RADIO;
DAL.MICROBIT_ID_RADIO_DATA_READY;
DAL.MICROBIT_ID_SERIAL;
DAL.MICROBIT_ID_THERMOMETER;
DAL.MICROBIT_INVALID_PARAMETER;
DAL.MICROBIT_IO_PINS;
DAL.MICROBIT_IO_PIN_SERVICE_DATA_SIZE;
DAL.MICROBIT_IO_PIN_SERVICE_PINCOUNT;
DAL.MICROBIT_LIGHT_SENSOR_AN_SET_TIME;
DAL.MICROBIT_LIGHT_SENSOR_CHAN_NUM;
DAL.MICROBIT_LIGHT_SENSOR_MAX_VALUE;
DAL.MICROBIT_LIGHT_SENSOR_MIN_VALUE;
DAL.MICROBIT_LIGHT_SENSOR_TICK_PERIOD;
DAL.MICROBIT_MAXIMUM_HEAPS;
DAL.MICROBIT_MULTI_BUTTON_ATTACHED;
DAL.MICROBIT_MULTI_BUTTON_HOLD_TRIGGERED_1;
DAL.MICROBIT_MULTI_BUTTON_HOLD_TRIGGERED_2;
DAL.MICROBIT_MULTI_BUTTON_STATE_1;
DAL.MICROBIT_MULTI_BUTTON_STATE_2;
DAL.MICROBIT_MULTI_BUTTON_SUPRESSED_1;
DAL.MICROBIT_MULTI_BUTTON_SUPRESSED_2;
DAL.MICROBIT_NAME_CODE_LETTERS;
DAL.MICROBIT_NAME_LENGTH;
DAL.MICROBIT_NOT_SUPPORTED;
DAL.MICROBIT_NO_DATA;
DAL.MICROBIT_NO_RESOURCES;
DAL.MICROBIT_NULL_DEREFERENCE;
DAL.MICROBIT_OK;
DAL.MICROBIT_OOM;
DAL.MICROBIT_PANIC_ERROR_CHARS;
DAL.MICROBIT_PIN_DEFAULT_SERVO_CENTER;
DAL.MICROBIT_PIN_DEFAULT_SERVO_RANGE;
DAL.MICROBIT_PIN_EVENT_NONE;
DAL.MICROBIT_PIN_EVENT_ON_EDGE;
DAL.MICROBIT_PIN_EVENT_ON_PULSE;
DAL.MICROBIT_PIN_EVENT_ON_TOUCH;
DAL.MICROBIT_PIN_EVT_FALL;
DAL.MICROBIT_PIN_EVT_PULSE_HI;
DAL.MICROBIT_PIN_EVT_PULSE_LO;
DAL.MICROBIT_PIN_EVT_RISE;
DAL.MICROBIT_PIN_MAX_OUTPUT;
DAL.MICROBIT_PIN_MAX_SERVO_RANGE;
DAL.MICROBIT_RADIO_BASE_ADDRESS;
DAL.MICROBIT_RADIO_DEFAULT_FREQUENCY;
DAL.MICROBIT_RADIO_DEFAULT_GROUP;
DAL.MICROBIT_RADIO_DEFAULT_TX_POWER;
DAL.MICROBIT_RADIO_EVT_DATAGRAM;
DAL.MICROBIT_RADIO_HEADER_SIZE;
DAL.MICROBIT_RADIO_MAXIMUM_RX_BUFFERS;
DAL.MICROBIT_RADIO_MAX_PACKET_SIZE;
DAL.MICROBIT_RADIO_PROTOCOL_DATAGRAM;
DAL.MICROBIT_RADIO_PROTOCOL_EVENTBUS;
DAL.MICROBIT_RADIO_STATUS_INITIALISED;
DAL.MICROBIT_SCHEDULER_RUNNING;
DAL.MICROBIT_SERIAL_DEFAULT_BAUD_RATE;
DAL.MICROBIT_SERIAL_DEFAULT_BUFFER_SIZE;
DAL.MICROBIT_SERIAL_EVT_DELIM_MATCH;
DAL.MICROBIT_SERIAL_EVT_HEAD_MATCH;
DAL.MICROBIT_SERIAL_EVT_RX_FULL;
DAL.MICROBIT_SERIAL_EVT_TX_EMPTY;
DAL.MICROBIT_SERIAL_IN_USE;
DAL.MICROBIT_SERIAL_RX_BUFF_INIT;
DAL.MICROBIT_SERIAL_RX_IN_USE;
DAL.MICROBIT_SERIAL_TX_BUFF_INIT;
DAL.MICROBIT_SERIAL_TX_IN_USE;
DAL.MICROBIT_STORAGE_BLOCK_SIZE;
DAL.MICROBIT_STORAGE_KEY_SIZE;
DAL.MICROBIT_STORAGE_MAGIC;
DAL.MICROBIT_STORAGE_SCRATCH_PAGE_OFFSET;
DAL.MICROBIT_STORAGE_STORE_PAGE_OFFSET;
DAL.MICROBIT_THERMOMETER_ADDED_TO_IDLE;
DAL.MICROBIT_THERMOMETER_EVT_UPDATE;
DAL.MICROBIT_THERMOMETER_PERIOD;
DAL.MICROBIT_UART_S_DEFAULT_BUF_SIZE;
DAL.MICROBIT_UART_S_EVT_DELIM_MATCH;
DAL.MICROBIT_UART_S_EVT_HEAD_MATCH;
DAL.MICROBIT_UART_S_EVT_RX_FULL;
DAL.MICROBIT_UART_S_EVT_TX_EMPTY;
DAL.MMA8653_CTRL_REG1;
DAL.MMA8653_CTRL_REG2;
DAL.MMA8653_CTRL_REG3;
DAL.MMA8653_CTRL_REG4;
DAL.MMA8653_CTRL_REG5;
DAL.MMA8653_DEFAULT_ADDR;
DAL.MMA8653_OUT_X_MSB;
DAL.MMA8653_SAMPLE_RANGES;
DAL.MMA8653_SAMPLE_RATES;
DAL.MMA8653_STATUS;
DAL.MMA8653_WHOAMI;
DAL.MMA8653_WHOAMI_VAL;
DAL.MMA8653_XYZ_DATA_CFG;
DAL.NORTH_EAST_DOWN;
DAL.NO_CONN;
DAL.NO_PWMS;
DAL.PIN_CAPABILITY_ANALOG;
DAL.PIN_CAPABILITY_DIGITAL;
DAL.PWM_PERSISTENCE_PERSISTENT;
DAL.PWM_PERSISTENCE_TRANSIENT;
DAL.RAW;
DAL.SIMPLE_CARTESIAN;
DAL.SYNC_SLEEP;
DAL.SYNC_SPINWAIT;
Delimiters.Colon;
Delimiters.Comma;
Delimiters.Dollar;
Delimiters.Fullstop;
Delimiters.Hash;
Delimiters.NewLine;
DigitalPin.P0;
DigitalPin.P10;
DigitalPin.P11;
DigitalPin.P12;
DigitalPin.P13;
DigitalPin.P14;
DigitalPin.P15;
DigitalPin.P16;
DigitalPin.P19;
DigitalPin.P1;
DigitalPin.P20;
DigitalPin.P2;
DigitalPin.P3;
DigitalPin.P4;
DigitalPin.P5;
DigitalPin.P6;
DigitalPin.P7;
DigitalPin.P8;
DigitalPin.P9;
Dimension.Strength;
Dimension.X;
Dimension.Y;
Dimension.Z;
Direction.Left;
Direction.Right;
DisplayMode.BackAndWhite;
DisplayMode.Greyscale;
EventBusSource.MES_BROADCAST_GENERAL_ID;
EventBusSource.MES_DEVICE_INFO_ID;
EventBusSource.MES_DPAD_CONTROLLER_ID;
EventBusSource.MES_SIGNAL_STRENGTH_ID;
EventBusSource.MICROBIT_ID_ACCELEROMETER;
EventBusSource.MICROBIT_ID_BUTTON_A;
EventBusSource.MICROBIT_ID_BUTTON_AB;
EventBusSource.MICROBIT_ID_BUTTON_B;
EventBusSource.MICROBIT_ID_GESTURE;
EventBusSource.MICROBIT_ID_IO_P0;
EventBusSource.MICROBIT_ID_IO_P10;
EventBusSource.MICROBIT_ID_IO_P11;
EventBusSource.MICROBIT_ID_IO_P12;
EventBusSource.MICROBIT_ID_IO_P13;
EventBusSource.MICROBIT_ID_IO_P14;
EventBusSource.MICROBIT_ID_IO_P15;
EventBusSource.MICROBIT_ID_IO_P16;
EventBusSource.MICROBIT_ID_IO_P19;
EventBusSource.MICROBIT_ID_IO_P1;
EventBusSource.MICROBIT_ID_IO_P20;
EventBusSource.MICROBIT_ID_IO_P2;
EventBusSource.MICROBIT_ID_IO_P3;
EventBusSource.MICROBIT_ID_IO_P4;
EventBusSource.MICROBIT_ID_IO_P5;
EventBusSource.MICROBIT_ID_IO_P6;
EventBusSource.MICROBIT_ID_IO_P7;
EventBusSource.MICROBIT_ID_IO_P8;
EventBusSource.MICROBIT_ID_IO_P9;
EventBusSource.MICROBIT_ID_RADIO;
EventBusValue.MES_ALERT_EVT_ALARM1;
EventBusValue.MES_ALERT_EVT_ALARM2;
EventBusValue.MES_ALERT_EVT_ALARM3;
EventBusValue.MES_ALERT_EVT_ALARM4;
EventBusValue.MES_ALERT_EVT_ALARM5;
EventBusValue.MES_ALERT_EVT_ALARM6;
EventBusValue.MES_ALERT_EVT_DISPLAY_TOAST;
EventBusValue.MES_ALERT_EVT_FIND_MY_PHONE;
EventBusValue.MES_ALERT_EVT_PLAY_RINGTONE;
EventBusValue.MES_ALERT_EVT_PLAY_SOUND;
EventBusValue.MES_ALERT_EVT_VIBRATE;
EventBusValue.MES_CAMERA_EVT_LAUNCH_PHOTO_MODE;
EventBusValue.MES_CAMERA_EVT_LAUNCH_VIDEO_MODE;
EventBusValue.MES_CAMERA_EVT_START_VIDEO_CAPTURE;
EventBusValue.MES_CAMERA_EVT_STOP_PHOTO_MODE;
EventBusValue.MES_CAMERA_EVT_STOP_VIDEO_CAPTURE;
EventBusValue.MES_CAMERA_EVT_STOP_VIDEO_MODE;
EventBusValue.MES_CAMERA_EVT_TAKE_PHOTO;
EventBusValue.MES_CAMERA_EVT_TOGGLE_FRONT_REAR;
EventBusValue.MES_DEVICE_DISPLAY_OFF;
EventBusValue.MES_DEVICE_DISPLAY_ON;
EventBusValue.MES_DEVICE_GESTURE_DEVICE_SHAKEN;
EventBusValue.MES_DEVICE_INCOMING_CALL;
EventBusValue.MES_DEVICE_INCOMING_MESSAGE;
EventBusValue.MES_DEVICE_ORIENTATION_LANDSCAPE;
EventBusValue.MES_DEVICE_ORIENTATION_PORTRAIT;
EventBusValue.MES_DPAD_BUTTON_1_DOWN;
EventBusValue.MES_DPAD_BUTTON_1_UP;
EventBusValue.MES_DPAD_BUTTON_2_DOWN;
EventBusValue.MES_DPAD_BUTTON_2_UP;
EventBusValue.MES_DPAD_BUTTON_3_DOWN;
EventBusValue.MES_DPAD_BUTTON_3_UP;
EventBusValue.MES_DPAD_BUTTON_4_DOWN;
EventBusValue.MES_DPAD_BUTTON_4_UP;
EventBusValue.MES_DPAD_BUTTON_A_DOWN;
EventBusValue.MES_DPAD_BUTTON_A_UP;
EventBusValue.MES_DPAD_BUTTON_B_DOWN;
EventBusValue.MES_DPAD_BUTTON_B_UP;
EventBusValue.MES_DPAD_BUTTON_C_DOWN;
EventBusValue.MES_DPAD_BUTTON_C_UP;
EventBusValue.MES_DPAD_BUTTON_D_DOWN;
EventBusValue.MES_DPAD_BUTTON_D_UP;
EventBusValue.MES_REMOTE_CONTROL_EVT_FORWARD;
EventBusValue.MES_REMOTE_CONTROL_EVT_NEXTTRACK;
EventBusValue.MES_REMOTE_CONTROL_EVT_PAUSE;
EventBusValue.MES_REMOTE_CONTROL_EVT_PLAY;
EventBusValue.MES_REMOTE_CONTROL_EVT_PREVTRACK;
EventBusValue.MES_REMOTE_CONTROL_EVT_REWIND;
EventBusValue.MES_REMOTE_CONTROL_EVT_STOP;
EventBusValue.MES_REMOTE_CONTROL_EVT_VOLUMEDOWN;
EventBusValue.MES_REMOTE_CONTROL_EVT_VOLUMEUP;
EventBusValue.MICROBIT_ACCELEROMETER_EVT_DATA_UPDATE;
EventBusValue.MICROBIT_BUTTON_EVT_CLICK;
EventBusValue.MICROBIT_EVT_ANY;
EventBusValue.MICROBIT_RADIO_EVT_DATAGRAM;
EventCreationMode.CreateAndFire;
EventCreationMode.CreateOnly;
Gesture.FreeFall;
Gesture.LogoDown;
Gesture.LogoUp;
Gesture.ScreenDown;
Gesture.ScreenUp;
Gesture.Shake;
Gesture.SixG;
Gesture.ThreeG;
Gesture.TiltLeft;
Gesture.TiltRight;
LedSpriteProperty.Blink;
LedSpriteProperty.Brightness;
LedSpriteProperty.Direction;
LedSpriteProperty.X;
LedSpriteProperty.Y;
Note.A3;
Note.A4;
Note.A5;
Note.A;
Note.B3;
Note.B4;
Note.B5;
Note.B;
Note.Bb3;
Note.Bb4;
Note.Bb5;
Note.Bb;
Note.C3;
Note.C4;
Note.C5;
Note.C;
Note.CSharp3;
Note.CSharp4;
Note.CSharp5;
Note.CSharp;
Note.D3;
Note.D4;
Note.D5;
Note.D;
Note.E3;
Note.E4;
Note.E5;
Note.E;
Note.Eb3;
Note.Eb4;
Note.Eb5;
Note.Eb;
Note.F3;
Note.F4;
Note.F5;
Note.F;
Note.FSharp3;
Note.FSharp4;
Note.FSharp5;
Note.FSharp;
Note.G3;
Note.G4;
Note.G5;
Note.G;
Note.GSharp3;
Note.GSharp4;
Note.GSharp5;
Note.GSharp;
NumberFormat.Int16BE;
NumberFormat.Int16LE;
NumberFormat.Int32BE;
NumberFormat.Int32LE;
NumberFormat.Int8BE;
NumberFormat.Int8LE;
NumberFormat.UInt16BE;
NumberFormat.UInt16LE;
NumberFormat.UInt8BE;
NumberFormat.UInt8LE;
PinPullMode.PullDown;
PinPullMode.PullNone;
PinPullMode.PullUp;
PulseValue.High;
PulseValue.Low;
Rotation.Pitch;
Rotation.Roll;
SerialPin.P0;
SerialPin.P12;
SerialPin.P13;
SerialPin.P14;
SerialPin.P15;
SerialPin.P16;
SerialPin.P1;
SerialPin.P2;
SerialPin.P8;
TouchPin.P0;
TouchPin.P1;
TouchPin.P2;
```

error:

```
Error:   -> Line 430 ('    bl -1003'), error: assembly error
   Maybe: bl    LABEL (uneven BL? at '-1003')
   Maybe: bls   LABEL (opcode name doesn't match at '<name>')

    at throwThumbErrors (C:\gh\pxt\built\pxt.js:66979:27)
    at assemble (C:\gh\pxt\built\pxt.js:66988:13)
    at Object.thumbEmit (C:\gh\pxt\built\pxt.js:67022:23)
    at finalEmit (C:\gh\pxt\built\pxt.js:63398:26)
    at catchErrors (C:\gh\pxt\built\pxt.js:65334:33)
    at Object.compileBinary (C:\gh\pxt\built\pxt.js:63244:17)
    at Object.compile (C:\gh\pxt\built\pxt.js:65729:38)
    at C:\gh\pxt\built\pxt.js:75803:36
    at tryCatcher (C:\gh\pxt\node_modules\bluebird\js\release\util.js:16:23)
    at Promise._settlePromiseFromHandler (C:\gh\pxt\node_modules\bluebird\js\release\promise.js:509:31)
    at Promise._settlePromise (C:\gh\pxt\node_modules\bluebird\js\release\promise.js:566:18)
    at Promise._settlePromise0 (C:\gh\pxt\node_modules\bluebird\js\release\promise.js:611:10)
    at Promise._settlePromises (C:\gh\pxt\node_modules\bluebird\js\release\promise.js:690:18)
    at Async._drainQueue (C:\gh\pxt\node_modules\bluebird\js\release\async.js:138:16)
    at Async._drainQueues (C:\gh\pxt\node_modules\bluebird\js\release\async.js:148:10)
    at Immediate.Async.drainQueues (C:\gh\pxt\node_modules\bluebird\js\release\async.js:17:14)
pxt_modules/core/dal.d.ts(1,1): error TS9200:   -> Line 430 ('    bl -1003'), error: assembly error
   Maybe: bl    LABEL (uneven BL? at '-1003')
   Maybe: bls   LABEL (opcode name doesn't match at '<name>')
```

--- a/pxtlib/emitter/emitter.ts
+++ b/pxtlib/emitter/emitter.ts
@@ -1367,7 +1367,7 @@ ${lbl}: .short 0xffff
                     }
                     ev = val + ""
                 }
-                if (/^\d+$/.test(ev))
+                if (/^[-]?\d+$/.test(ev))
                     return ir.numlit(parseInt(ev));
                 return ir.rtcall(ev, [])
             } else if (decl.kind == SK.PropertySignature) {

--- a/pxtlib/emitter/emitter.ts
+++ b/pxtlib/emitter/emitter.ts
@@ -1367,7 +1367,7 @@ ${lbl}: .short 0xffff
                     }
                     ev = val + ""
                 }
-                if (/^[-]?\d+$/.test(ev))
+                if (/^[+-]?\d+$/.test(ev))
                     return ir.numlit(parseInt(ev));
                 return ir.rtcall(ev, [])
             } else if (decl.kind == SK.PropertySignature) {

